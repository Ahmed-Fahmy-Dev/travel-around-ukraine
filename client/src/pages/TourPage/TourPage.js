import React from 'react';
import { Box, Container, Typography, Stack } from '@mui/material';
// Icons
import PersonIcon from '@mui/icons-material/Person';
import BedIcon from '@mui/icons-material/Bed';

import { TourAccordion, TourReasonToChoose, TourInfoDialog } from '../../components';
import styles from './TourPage.module.scss';

const included = [
  { icon: <PersonIcon color="primary" />, service: 'Professional guide' },
  { icon: <BedIcon color="primary" />, service: 'Accomodation' },
];
const cost = {
  eur: 70,
  usd: 70,
  uah: 2584,
};
const dates = { beginDate: new Date('2022-02-01'), endDate: new Date('2022-02-25') };

const details = {
  duration: '3 days',
  departs: 'First city',
  returns: 'Second city',
};

const TourPage = () => (
  <Container>
    <div className={styles.content}>
      <Box component="aside" className={styles.aside}>
        <TourInfoDialog included={included} cost={cost} dates={dates} details={details} />
      </Box>
      <Box sx={{ flex: 1 }}>
        <Typography variant="h2">About us</Typography>
        <Box sx={{ paddingBottom: 3 }}>
          <Typography>
            Each medieval city is full of various legends and mystical stories. Lviv is no exception because more than
            760 years of history simply could not help but leave mysterious legends.
          </Typography>

          <Typography>
            The guide will tell and show that Lviv legends can impress no less than Lviv architecture. This tour will be
            filled with incredible stories - unusual stories from the life of the Galician capital.
          </Typography>

          <Typography>
            How can one explain the black colour of one of the houses on Rynok Square? How did Adam Senyavsky marry his
            daughter? How did the first bulletin board appear in Lviv? Why did one Lviv lady make a not entirely decent
            offer to the tram? Are the legends about Lviv ghosts true? What do Lviv courtyards hide?
          </Typography>

          <Typography>All this and more you can learn during this tour.</Typography>

          <Typography>Looks like today. The program of individual excursions: at any hour.</Typography>
        </Box>
        <TourAccordion title="Reasons to choose our tour">
          <Stack direction="row" gap={3} flexWrap="wrap" mt={2} pl={2}>
            <TourReasonToChoose
              number={1}
              description="Lorem ipsum dolor sit amet consectetur, adipisicing elit. Nisi ducimus tenetur in aspernatur, asperiores."
            />
            <TourReasonToChoose number={2} description="Lorem ipsum dolor sit amet consectetur, adipisicing elit." />
            <TourReasonToChoose
              number={3}
              description="Lorem ipsum dolor sit amet consectetur, adipisicing elit. Nisi ducimus tenetur in aspernatur, asperiores."
            />
          </Stack>
        </TourAccordion>
        <TourAccordion title="What is included?">
          <Stack direction="row" gap="20px">
            {included.map(({ icon, service }) => (
              <Stack key={service} direction="row" gap="5px" alignItems="center" flexWrap="wrap">
                {icon}
                <Typography gutterBottom={false}>{service}</Typography>
              </Stack>
            ))}
          </Stack>
        </TourAccordion>
      </Box>
    </div>
  </Container>
);

export default TourPage;
