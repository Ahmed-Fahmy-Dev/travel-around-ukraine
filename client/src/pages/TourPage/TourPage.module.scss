.content {
  display: flex;
  gap: 20px;
}

.aside {
  max-width: 370px;
  width: 100%;

	@media (max-width: '1050px') {
		display: none;
	}
}