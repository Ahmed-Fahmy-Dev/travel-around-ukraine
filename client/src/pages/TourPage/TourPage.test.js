import { render } from '@testing-library/react';
import TourPage from './TourPage';

import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils';

describe('Tour Page snapshot testing', () => {
  test('should Tour Page match snapshot', () => {
    const { asFragment } = render(<TourPage />);
    mockAllIsIntersecting(true);

    expect(asFragment()).toMatchSnapshot();
  });
});
