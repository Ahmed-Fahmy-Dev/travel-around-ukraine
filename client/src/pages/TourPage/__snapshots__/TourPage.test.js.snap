// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tour Page snapshot testing should Tour Page match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-btfqat"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <nav
        class="css-rtewkn"
      >
        <div
          class="css-yefzdd-MuiStack-root"
        >
          <a
            class="MuiTypography-root MuiTypography-body1 MuiLink-root MuiLink-underlineNone css-3ehev0-MuiTypography-root-MuiLink-root"
            href="#about-tour"
          >
            About us
          </a>
          <a
            class="MuiTypography-root MuiTypography-body1 MuiLink-root MuiLink-underlineNone css-3ehev0-MuiTypography-root-MuiLink-root"
            href="#reasons-to-choose"
          >
            Reasons to choose
          </a>
          <a
            class="MuiTypography-root MuiTypography-body1 MuiLink-root MuiLink-underlineNone css-3ehev0-MuiTypography-root-MuiLink-root"
            href="#included"
          >
            What is included?
          </a>
        </div>
      </nav>
    </div>
  </div>
  <div
    class="MuiBox-root css-1vbcp5u"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="css-1vlser0-MuiStack-root"
      >
        <div
          class="MuiBox-root css-1rr4qq7"
        >
          <div
            class="MuiBox-root css-e2d11k"
            id="about-tour"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
            >
              About tour
            </h2>
            <div
              class="MuiBox-root css-hshm0p"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Each medieval city is full of various legends and mystical stories. Lviv is no exception because more than 760 years of history simply could not help but leave mysterious legends.
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                The guide will tell and show that Lviv legends can impress no less than Lviv architecture. This tour will be filled with incredible stories - unusual stories from the life of the Galician capital.
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                How can one explain the black colour of one of the houses on Rynok Square? How did Adam Senyavsky marry his daughter? How did the first bulletin board appear in Lviv? Why did one Lviv lady make a not entirely decent offer to the tram? Are the legends about Lviv ghosts true? What do Lviv courtyards hide?
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                All this and more you can learn during this tour.
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Looks like today. The program of individual excursions: at any hour.
              </p>
            </div>
          </div>
          <div
            class="MuiBox-root css-e2d11k"
            id="reasons-to-choose"
          >
            <div>
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root Mui-expanded MuiAccordion-gutters css-i5ltjc-MuiPaper-root-MuiAccordion-root"
              >
                <div
                  aria-expanded="true"
                  class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded MuiAccordionSummary-gutters css-tsm144-MuiButtonBase-root-MuiAccordionSummary-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiAccordionSummary-content Mui-expanded MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
                  >
                    <h2
                      class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
                    >
                      Reasons to choose our tour
                    </h2>
                  </div>
                  <div
                    class="MuiAccordionSummary-expandIconWrapper Mui-expanded css-yw020d-MuiAccordionSummary-expandIconWrapper"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                      data-testid="ExpandMoreIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
                  style="min-height: 0px;"
                >
                  <div
                    class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                  >
                    <div
                      class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                    >
                      <div
                        class="MuiAccordion-region"
                        role="region"
                      >
                        <div
                          class="MuiAccordionDetails-root css-m733ju-MuiAccordionDetails-root"
                        >
                          <div
                            class="css-1ytg1c4-MuiStack-root"
                          >
                            <div
                              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-9l9va7-MuiPaper-root"
                            >
                              <div
                                class="MuiBox-root css-1f42jzd"
                              />
                              <div
                                class="MuiBox-root css-79elbk"
                              >
                                <p
                                  class="MuiTypography-root MuiTypography-body1 css-sliiol-MuiTypography-root"
                                >
                                  1
                                </p>
                                <p
                                  class="MuiTypography-root MuiTypography-body1 css-ykwshg-MuiTypography-root"
                                >
                                  Lorem ipsum dolor sit amet consectetur, adipisicing elit. Nisi ducimus tenetur in aspernatur, asperiores.
                                </p>
                              </div>
                            </div>
                            <div
                              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-9l9va7-MuiPaper-root"
                            >
                              <div
                                class="MuiBox-root css-1f42jzd"
                              />
                              <div
                                class="MuiBox-root css-79elbk"
                              >
                                <p
                                  class="MuiTypography-root MuiTypography-body1 css-sliiol-MuiTypography-root"
                                >
                                  2
                                </p>
                                <p
                                  class="MuiTypography-root MuiTypography-body1 css-ykwshg-MuiTypography-root"
                                >
                                  Lorem ipsum dolor sit amet consectetur, adipisicing elit.
                                </p>
                              </div>
                            </div>
                            <div
                              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-9l9va7-MuiPaper-root"
                            >
                              <div
                                class="MuiBox-root css-1f42jzd"
                              />
                              <div
                                class="MuiBox-root css-79elbk"
                              >
                                <p
                                  class="MuiTypography-root MuiTypography-body1 css-sliiol-MuiTypography-root"
                                >
                                  3
                                </p>
                                <p
                                  class="MuiTypography-root MuiTypography-body1 css-ykwshg-MuiTypography-root"
                                >
                                  Lorem ipsum dolor sit amet consectetur, adipisicing elit. Nisi ducimus tenetur in aspernatur, asperiores.
                                </p>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-e2d11k"
            id="included"
          >
            <div>
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root Mui-expanded MuiAccordion-gutters css-i5ltjc-MuiPaper-root-MuiAccordion-root"
              >
                <div
                  aria-expanded="true"
                  class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded MuiAccordionSummary-gutters css-tsm144-MuiButtonBase-root-MuiAccordionSummary-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiAccordionSummary-content Mui-expanded MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
                  >
                    <h2
                      class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
                    >
                      What is included?
                    </h2>
                  </div>
                  <div
                    class="MuiAccordionSummary-expandIconWrapper Mui-expanded css-yw020d-MuiAccordionSummary-expandIconWrapper"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                      data-testid="ExpandMoreIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
                  style="min-height: 0px;"
                >
                  <div
                    class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                  >
                    <div
                      class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                    >
                      <div
                        class="MuiAccordion-region"
                        role="region"
                      >
                        <div
                          class="MuiAccordionDetails-root css-m733ju-MuiAccordionDetails-root"
                        >
                          <div
                            class="css-l0jt21-MuiStack-root"
                          >
                            <div
                              class="css-11wm0nl-MuiStack-root"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                                data-testid="PersonIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                                />
                              </svg>
                              <p
                                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                              >
                                Professional guide
                              </p>
                            </div>
                            <div
                              class="css-11wm0nl-MuiStack-root"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                                data-testid="BedIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M21 10.78V8c0-1.65-1.35-3-3-3h-4c-.77 0-1.47.3-2 .78-.53-.48-1.23-.78-2-.78H6C4.35 5 3 6.35 3 8v2.78c-.61.55-1 1.34-1 2.22v6h2v-2h16v2h2v-6c0-.88-.39-1.67-1-2.22zM14 7h4c.55 0 1 .45 1 1v2h-6V8c0-.55.45-1 1-1zM5 8c0-.55.45-1 1-1h4c.55 0 1 .45 1 1v2H5V8z"
                                />
                              </svg>
                              <p
                                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                              >
                                Accomodation
                              </p>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1y1dfia"
      >
        <div
          class="css-3y1137-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1q01i3p-MuiTypography-root"
          >
            €70
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation css-fwdzms-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            More info
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1bev9rf"
      style="webkit-transform: translateY(NaNpx); transform: translateY(NaNpx); webkit-transition: -webkit-transform 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms; transition: transform 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms;"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div
          class="css-3y1137-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1q01i3p-MuiTypography-root"
          >
            €70
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation css-fwdzms-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            More info
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
